### PR TITLE
feat: add `join` function

### PR DESCRIPTION
**Pull request by 6h15m**
#8

### CODE REVIEW

**Review by ppeeou — APPROVED** [review]
LGTM

File: type-check/join.test.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+const { checks, check } = Test;
+
+const res1 = join("-", []);
+const res2 = join("!", [1, 2, 3, 4]);
+const res3 = join("*", toAsync([1, 2, 3, 4]));
+const res4 = pipe([1, 2, 3, 4], join("-"));
+const res5 = pipe(toAsync([1, 2, 3, 4]), join("#"));
+
+checks([
+  check<typeof res1, string, Test.Pass>(),
```
Comment (shine1594) [inline type-check/join.test.ts:13]: It seems like `res1` should be inferred as an empty array.

File: type-check/join.test.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+const { checks, check } = Test;
+
+const res1 = join("-", []);
+const res2 = join("!", [1, 2, 3, 4]);
+const res3 = join("*", toAsync([1, 2, 3, 4]));
+const res4 = pipe([1, 2, 3, 4], join("-"));
+const res5 = pipe(toAsync([1, 2, 3, 4]), join("#"));
+
+checks([
+  check<typeof res1, [], Test.Pass>(),
```
Comment (shine1594) [inline type-check/join.test.ts:13]: `join` only returns string. Type of `res1` must be empty string, not an empty array.
```suggestion
heck<typeof res1, "", Test.Pass>(),
```

**Review by hg-pyun — APPROVED** [review]
LGTM

File: type-check/join.test.ts (hunk 2 above)
Comment (6h15m) [reply type-check/join.test.ts]: I fixed it :>

**Review by shine1594 — APPROVED** [review]

**Pull request merged**